tools/type-checking/type-check.py

```diff
@@ -484,6 +485,39 @@ def ci(self):
             print(self._red("✗ Some checks failed, see above for errors"))
             return False
 
+    def _check_strictness_file_sorted(self) -> bool:
+        """Check that each section of the strictness file is sorted.
+
+        Returns True if all sections are sorted, False otherwise.
+        """
+        with open(self.strictness_config_path, "r") as f:
+            config: dict[str, list[str]] = json.load(f)
+
+        all_sorted = True
+        for level_str, patterns in config.items():
+            if level_str == "comment":
+                continue
+
+            # Check if this section is sorted
+            sorted_patterns = sorted(patterns)
+            if patterns != sorted_patterns:
+                all_sorted = False
+                print(
+                    f"{self.error} Section '{level_str}' is not sorted in {self.strictness_config_path}:"
+                )
+
+                # Show which entries are out of order
+                for i, (actual, expected) in enumerate(zip(patterns, sorted_patterns)):
+                    if actual != expected:
```

Comment:
Isn't this kinda noisy aka a rather poor diff.

For:

A A
B F
C B
D C
E D
F E

You'll get almost every line printed here? Maybe break after the first diff?